dom/src/main/java/org/apache/james/mime4j/message/MessageBuilder.java

```diff
@@ -75,15 +75,68 @@
     private BodyFactory bodyFactory;
     private boolean flatMode;
     private boolean rawContent;
+    private final boolean ensureDateNotNull;
```

Comment:
Why all this deprecation logic while you can keep this value true by default

and expose a method .notOverrideNullDate() that sets this parameter to false.

Thus we avoid most of the troubble. Without deprecation.